src/openzaak/utils/inclusion.py

```diff
@@ -0,0 +1,222 @@
+            # serializer
+            nested_serializers = {
+                field_name[len(name) + 1 :]: serializer
+                for field_name, serializer in inclusion_serializers.items()
+                if field_name.startswith(name)
+            }
+
+            nested_path = (
+                new_path if self.nested_inclusions_use_complete_path else new_path[:-1]
+            )
+            for entry in self._instance_inclusions(
+                nested_path,
+                inclusion_serializer(instance=object),
+                obj,
+                inclusion_serializers=nested_serializers,
+            ):
+                yield entry
+
+    def _some_related_field_inclusions(
+        self, path, field, instance, inclusion_serializer
+    ):
+        if self.allowed_paths is not None and path not in self.allowed_paths:
+            return []
+
+        # Properly include loose fk fields
+        if (
+            isinstance(field, FKOrURLField)
+            or isinstance(field, LooseFKHyperlinkedRelatedField)
+            or isinstance(field, LooseFKHyperlinkedIdentityField)
+        ):
+            return self._loose_fk_field_inclusions(
+                path, field, instance, inclusion_serializer
+            )
+        elif isinstance(field, HyperlinkedRelatedField):
+            return self._primary_key_related_field_inclusions(
+                path, field, instance, inclusion_serializer
+            )
+        return super()._some_related_field_inclusions(
+            path, field, instance, inclusion_serializer
+        )
+
+    def _loose_fk_field_inclusions(self, path, field, instance, inclusion_serializer):
+        value = field.get_attribute(instance)
+        # In case it's an external resource
+        if isinstance(value, str) or hasattr(value, "_initial_data"):
+            if self._has_been_seen_external(value):
+                return
+
+            try:
+                yield getattr(instance, field.field_name)  # ._initial_data
+            except FetchError:  # Something failed during fetching, ignore this instance
+                return []
+        else:
+            for entry in self._primary_key_related_field_inclusions(
+                path, field, instance, inclusion_serializer
+            ):
+                yield entry
+
+
+class InclusionJSONRenderer(_InclusionJSONRenderer, CamelCaseJSONRenderer):
+    """
+    Ensure that the InclusionJSONRenderer produces camelCase and properly loads loose fk
+    objects
+    """
+
+    loader_class = InclusionLoader
+    response_results_key = "results"
+    response_data_key = "data"
+
+    @use_connection_pool
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        # Only render inclusions for list operation
+        if renderer_context["view"].action == "list":
+            return super().render(data, accepted_media_type, renderer_context)
+        return super(_InclusionJSONRenderer, self).render(
```

Comment:
that `super()` call is throwing me off a bit - wouldn't that result in a non-camel-case `JSONRenderer` being used?

Reply:
you're right, I've moved this part to the library and now just override to add the connection pooling